### PR TITLE
Add SlackAPIFileOperator impementing files.upload from Slack API

### PR DESCRIPTION
**Pull request by mikaeld**
Implementation of files.upload from Slack API. This operator will send files to any Slack channel.

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by mik-laj** [comment]
@mikaeld  Can you aad tests?

**Comment by mikaeld** [comment]
I looked for SlackAPIOperator test cases; I couldn't find any. Can you point me in the right direction if there are existing tests? I could use those as a starting point, as I've only ever used pytest for tests and I'm not yet familiar with the mocks required in Airflow.

**Comment by mik-laj** [comment]
Tests for Slack operators:
https://github.com/apache/airflow/blob/master/tests/providers/slack/operators/test_slack.py
Tests for Slack's hook:
https://github.com/apache/airflow/blob/master/tests/providers/slack/hooks/test_slack.py
Guide about testing: 
https://github.com/apache/airflow/blob/master/TESTING.rst
Guide about static checks:
https://github.com/apache/airflow/blob/master/STATIC_CODE_CHECKS.rst
Contributor guide:
https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst
I learned about mock from the official documentation
https://docs.python.org/3/library/unittest.mock.html

Can I help with anything else?

**Comment by mikaeld** [comment]
Thanks for all the resources, I'll have a look at those!

File: airflow/providers/slack/operators/slack.py [hunk 1]
```diff
@@ -151,3 +151,65 @@ def construct_api_call_params(self):
...
+
+    Examples:
+
+    .. code-block:: python
+
+        slack = SlackAPIFileOperator(
+            task_id="slack_file_upload",
+            dag=dag,
+            conn_id="slack",
+            channel="general",
```
Comment (kaxil) [inline airflow/providers/slack/operators/slack.py]:
```suggestion
            channel="#general",
```

File: airflow/providers/slack/operators/slack.py [hunk 2]
```diff
@@ -151,3 +151,65 @@ def construct_api_call_params(self):
...
+    Send a file to a slack channel
+
+    Examples:
+
+    .. code-block:: python
+
+        slack = SlackAPIFileOperator(
+            task_id="slack_file_upload",
+            dag=dag,
+            conn_id="slack",
```
Comment (kaxil) [inline airflow/providers/slack/operators/slack.py]:
```suggestion
            slack_conn_id="slack",
```

**Review by kaxil — COMMENTED** [review]
2 minor comments other LGTM

**Review by turbaszek — APPROVED** [review]
LGTM once Kaxil's comments are addressed

**Pull request merged**